RevenueCatUI/CustomerCenter/View+PresentCustomerCenter.swift

```diff
@@ -0,0 +1,124 @@
+//      https://opensource.org/licenses/MIT
+//
+//  View+PresentCustomerCenter.swift
+//
+//  Created by Toni Rico Diez on 2024-07-15.
+
+import RevenueCat
+import SwiftUI
+
+#if !os(macOS) && !os(tvOS)
+
+/// Presentation options to use with the [presentCustomerCenter](x-source-tag://presentCustomerCenter) View modifiers.
+public enum CustomerCenterPresentationMode {
+
+    /// Customer center presented using SwiftUI's `.sheet`.
+    case sheet
+
+    /// Customer center presented using SwiftUI's `.fullScreenCover`.
+    case fullScreen
+
+}
+
+extension CustomerCenterPresentationMode {
+
+    // swiftlint:disable:next missing_docs
+    public static let `default`: Self = .sheet
+
+}
+
+// swiftlint:disable file_length
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable, message: "RevenueCatUI does not support macOS yet")
+@available(tvOS, unavailable, message: "RevenueCatUI does not support tvOS yet")
+extension View {
+
+    /// Presents the ``CustomerCenter``.
+    /// Example:
+    /// ```swift
+    /// var body: some View {
+    ///    YourApp()
+    ///      .presentCustomerCenter()
+    /// }
+    /// ```
+
+    /// - Parameter isPresented: Binding indicating whether the customer center should be displayed
+    /// - Parameter onDismiss: Callback executed when the customer center wants to be dismissed.
+    /// Make sure you stop presenting the customer center when this is called
+    /// - Parameter presentationMode: The desired presentation mode of the customer center. Defaults to `.sheet`.
+    public func presentCustomerCenter(
+        isPresented: Binding<Bool>,
+        onDismiss: @escaping () -> Void,
+        presentationMode: CustomerCenterPresentationMode = .default
+    ) -> some View {
+        return self.modifier(PresentingCustomerCenterModifier(
+            isPresented: isPresented,
+            onDismiss: onDismiss,
+            myAppPurchaseLogic: nil,
+            presentationMode: presentationMode
+        ))
+    }
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+private struct PresentingCustomerCenterModifier: ViewModifier {
+
+    var presentationMode: CustomerCenterPresentationMode
+    var onDismiss: (() -> Void)
+
+    init(
+        isPresented: Binding<Bool>,
+        onDismiss: @escaping () -> Void,
+        myAppPurchaseLogic: MyAppPurchaseLogic?,
+        presentationMode: CustomerCenterPresentationMode,
+        purchaseHandler: PurchaseHandler? = nil
+    ) {
+        self._isPresented = isPresented
+        self.presentationMode = presentationMode
+        self.onDismiss = onDismiss
+        self._purchaseHandler = .init(wrappedValue: purchaseHandler ??
+                                      PurchaseHandler.default(performPurchase: myAppPurchaseLogic?.performPurchase,
+                                                              performRestore: myAppPurchaseLogic?.performRestore))
```

Comment:
I added this from paywalls, but removed it from the public API for now, since we still don't allow passing it to the `CustomerCenterView`